kollect-core/src/main/kotlin/kollect/Cache.kt

```diff
@@ -0,0 +1,45 @@
+package kollect
+
+import arrow.core.Option
+import arrow.core.Tuple2
+
+/**
+ * A `Cache` interface so the users of the library can provide their own cache.
+ */
+interface DataSourceCache {
```

Comment:
At some point we may want to turn this into a type class parametric to `F` because this assumes now that a `Cache` will return its results blocking if it needs to go async to fetch them and that is not good.

Reply:
I've created the ticket in `fetch` (https://github.com/47deg/fetch/issues/148). I will create the ticket here once this gets merged